### PR TITLE
add config for sensor gc4023.

### PR DESCRIPTION
**Pull request by duchengyao**
#1003 

Need uncomment 
 https://github.com/OpenIPC/openingenic/blob/04774c2d3f043b91baab1a5260560ec1a8cf6101/kernel/Kbuild#L25

### CODE REVIEW

**Review by ZigFisher — COMMENTED** [review]
Good afternoon, thank you for the PR
Please show the full information from the `ipctool` utility

**Comment by duchengyao** [comment]
majestic:
```
19:25:34  <       majestic> [sns_conf] try_sensor_config@25          sensor autodetection failed
19:25:34  <       majestic> [sns_conf] try_sensor_config@26          Use SENSOR environment variable to specify it explicitly.
```

I must set sensor manually using `fw_setenv sensor "gc4023"`. Then, sensor works.

```
root@openipc-t31:~# ./ipctool-mips32 
---
chip:
  vendor: Ingenic
  model: T31X
board:
  vendor: OpenIPC
  version: 2.3.09.24
ethernet:
  mac: "da:12:da:a5:52:ae"
rom:
- type: nor
  block: 32K
  partitions:
    - name: boot
      size: 0x40000
      sha1: 1d72f276
    - name: env
      size: 0x10000
      sha1: 368676b8
      contains:
        - name: uboot-env
          offset: 0x0
    - name: kernel
      size: 0x200000
      sha1: 2d09c73a
    - name: rootfs
      size: 0x500000
      path: /,squashfs
      sha1: 52c82ac5
    - name: rootfs_data
      size: 0xb0000
      path: /overlay,jffs2,rw
  size: 8M
ram:
  total: 64M
  media: 64M
firmware:
  kernel: "3.10.14__isvp_swan_1.0__ (PREEMPT Sun Sep 24 00:19:38 CST 2023)"
  toolchain: buildroot-gcc-12.2.0
```

**Comment by ZigFisher** [comment]
Unfortunately, until sensor detection is available we won't be able to accept PRs. 
I see that there is a discussion about this in the OpenIPC_EN group and therefore I wish you success in the configuration. 
Please let me know afterwards when all issues are resolved.

File: general/package/ingenic-osdrv-t31/files/sensor/gc4023.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+sensor:
+  name: gc4023
+  address: 0x21
+  width: 2560
+  height: 1440
+  bus: i2c
+  resolutions: 2560x1440x25_400,1920x1080x60_300,1280x720x60_100
```
Comment (dimerr) [inline general/package/ingenic-osdrv-t31/files/sensor/gc4023.yaml:7]: the only supported resolution is 2560x1440x25_400

**Comment by duchengyao** [comment]
```
root@openipc-t31:~# ./ipctool-mips32 
---
chip:
  vendor: Ingenic
  model: T31X
board:
  vendor: OpenIPC
  version: 2.3.09.24
ethernet:
  mac: "72:16:a3:e5:8e:9a"
rom:
- type: nor
  block: 32K
  partitions:
    - name: boot
      size: 0x40000
      sha1: 1d72f276
    - name: env
      size: 0x10000
      sha1: 6bcb0ac4
      contains:
        - name: uboot-env
          offset: 0x0
    - name: kernel
      size: 0x200000
      sha1: 263a5526
    - name: rootfs
      size: 0x500000
      path: /,squashfs
      sha1: 8c178c28
    - name: rootfs_data
      size: 0xb0000
      path: /overlay,jffs2,rw
  size: 8M
ram:
  total: 64M
  media: 64M
firmware:
  kernel: "3.10.14__isvp_swan_1.0__ (PREEMPT Sun Sep 24 17:20:58 CST 2023)"
  toolchain: buildroot-gcc-12.2.0
sensors:
- vendor: GalaxyCore
  model: GC4023
  control:
    bus: 0
    type: i2c
    addr: 0x42
```

**Review by FlyRouter — APPROVED** [review]
Thank you

**Pull request merged**